### ISSUE TITLE
Suboptimal play

### ISSUE THREAD

**Issue by Ardweaden** [niklasf/syzygy-tables.info#5]
In certain positions the suggested best move is far worse than the optimal move. For example, this position: 8/8/8/6kp/6p1/6K1/7R/r7 w - - 1 2 yields "bestmove": "h2h4" which loses rook in 2 moves. According to Nalimov tablebases, it drops from mate in 22 to mate in 8.

**Comment by niklasf** [comment]
This is by design. Syzygy tablebases use the DTZ (distance to zeroing) metric instead of DTM (depth to mate).

For winning positions DTZ aims to reset the counter of the 50-move drawing rule as quickly as possible (by a pawn move or capture) while maintaining the win - even if that leads to a longer mate or loss of material. Imagine you wasted a lot of moves and the 50-move limit is getting very close. Now the shortest mate may not be short enough. Instead it could be better to first sacrifice some material to reset the counter.

Conversly, imagine you defended the endgame in your example tenaciously and white only has 1 move left until you can claim a 50-move draw. You can draw by preventing any pawn move and capture. So in that case Rh4 is indeed the best move.

*tl;dr: Syzygy tables aim for optimal outcome under the 50-move rule (which is all that matters in practical play), not shortest mate.*
